### PR TITLE
test: Verify KMS settings are not stored if the checkbox is not set COMPASS-5890

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

Verify KMS settings are not stored if the checkbox is not set.
Follow up PR on https://github.com/mongodb-js/compass/pull/3125

We skip the test for Linux CI since there is an issue with displaying saved connections for these platforms. The corresponding ticket: https://jira.mongodb.org/browse/COMPASS-5911

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx [hunk 1]
```diff
@@ -157,11 +157,11 @@ function CSFLETab({
...
             handleStoreCredentialsChanged(event.target.checked);
           }}
-          data-testid="csfle-storeCredentials-input"
-          id="csfle-storeCredentials-input"
+          data-testid="csfle-store-credentials-input"
+          id="csfle-store-credentials-input"
           label={
             <>
-              <Label htmlFor="csfle-storeCredentials-input">
+              <Label htmlFor="csfle-store-credentials-inputt">
```
Comment (lerouxb) [inline packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx:164]: Minor thing, but inputt seems like a typo?

File: packages/compass-e2e-tests/tests/instance-sidebar.test.ts [hunk 2]
```diff
@@ -176,7 +176,7 @@ describe('Instance sidebar', function () {
     );
 
     await browser.scrollToVirtualItem(
-      Selectors.SidebarDatabaseAndConnectionList,
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/instance-sidebar.test.ts:179]: oops :)

File: packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts [hunk 3]
```diff
@@ -313,6 +334,19 @@ async function getValue(
   return value || null;
 }
 
+async function getAceValue(
```
Comment (mcasimir) [inline packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts:337]: silly nit, would it be better to name this `getEditorValue`, anticipating us replacing Ace eventually?

Comment (alenakhineika) [reply packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts:337]: I made it consistent with `browser.setAceValue()` that is already in use.

**Review by lerouxb — APPROVED** [review]

File: packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts [hunk 4]
```diff
@@ -191,6 +191,26 @@ export async function getConnectFormState(
     ),
   });
 
+  // FLE2
```
Comment (addaleax) [inline packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts:194]: Side note that is totally irrelevant for review: I am not sure what to do with our internal usage of “CSFLE” (as an umbrella term for CSFLE+QE) and “FLE2”. It’s just naming, and it’s all internal, but should we worry about people down the line being confused by our outdated terminology? Is it worth doing a huge renaming of everything?

File: packages/compass-e2e-tests/tests/in-use-encryption.test.ts [hunk 5]
```diff
@@ -7,35 +7,58 @@ import { MONGODB_VERSION } from '../helpers/compass';
...
+    const databaseName = 'fle-test';
+    const collectionName = 'my-another-collection';
     let compass: Compass;
     let browser: CompassBrowser;
 
     before(async function () {
+      if (
+        semver.lt(MONGODB_VERSION, '4.2.20') ||
+        process.env.MONGODB_USE_ENTERPRISE !== 'yes' ||
+        process.platform === 'linux'
```
Comment (addaleax) [inline packages/compass-e2e-tests/tests/in-use-encryption.test.ts:48]: Is there a ticket for this? If so, can you link it in a comment (and link the ticket as related to the one for this PR)? `// TODO(COMPASS-XXXX): Fix this on Linux`

File: packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts (hunk 4 above)
Comment (alenakhineika) [reply packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts:194]: Absolutely agree that we need to unify the name!! For me this is also confusing.

Comment (alenakhineika) [reply packages/compass-e2e-tests/helpers/commands/get-connect-form-state.ts:194]: Thx! ✨

**Pull request merged**